library/lua/gui/widgets.lua

```diff
@@ -415,6 +427,11 @@ Label.ATTRS{
     on_click = DEFAULT_NIL,
     on_rclick = DEFAULT_NIL,
     scroll_keys = STANDARDSCROLL,
+    --
+    show_scroll_icons = true,
```

Comment:
maybe default this to `DEFAULT_NIL` and allow it to autodetect whether the content height exceeds the body height

Reply:
and if the caller wants to force them on or off, they can set `show_scroll_icons` to a boolean

Reply:
The actual icons are only shown when there is a need for them. I.e. they won't necessarily be visible even with `show_scroll_icons = true` right now. What it will do is potentially set that `frame_inset.r` to `1`. I think forcing the arrows to be always visible will be confusing.

Maybe `DEFAULT_NIL` could set the boolean value on widget creation but before adjusting the `frame_inset.r`?

Reply:
fair enough -- defaulting to true is fine if the behavior is still dynamic.

On that note, the length of the text is dynamic too, so we have to be careful about just doing things on init. I *believe* the vertical height of the text can only change when `setText()` is called since dynamic text can't add lines -- they can only change the contents of existing lines. This is because newlines returned from text-generating functions don't cause the rendered text to go to the next line. I'm on the fence on whether this is a bug or intended behavior, but it's the current implementation.

There should also be a change when you can no longer scroll in a particular direction. Like maybe changing the icon pen from COLOR_CYAN to something darker (or gray?)

Reply:
>There should also be visual feedback when you can no longer scroll in a particular direction. Like maybe changing the icon pen from COLOR_CYAN to something darker (or gray?)

Currently, the change is the icon disappears. In other words, a scroll icon is visible if and only if it is possible to scroll in indicated direction.

Reply:
I've made some changes.

* since `render_scroll_icons` only works with a label, it's now a class function
* `show_scroll_icons` has 4 possible values: `false` for no icons, `left` for icons on the first column on the left (also ensuring `frame_inset.l >= 1`), `right` - last column on the right, `DEFAULT_NIL` - same as `right` if text height greater than `frame_body.height`, else same as `false`.

We can't call `Label:update_scroll_inset` from `Label:setText`, because `self.frame_body` isn't calculated at the time it's called (this is only needed for the `DEFAULT_NIL` logic).